cmk/special_agents/agent_fuse_management_central.py

```diff
@@ -0,0 +1,265 @@
+    env_alerts: List[Alert] = []
+    if "environments" not in layout:
+        return env_alerts
+
+    for environment in layout["environments"]:
+        for component in environment["componentTypes"]:
+            env_alert: Alert = Alert(
+                environment["id"], environment["name"], "", component["displayName"], 0, 0, ""
+            )
+            if (environment["id"], component["id"]) in env_alert_map:
+                alert = env_alert_map[(environment["id"], component["id"])]
+                if "errors" in alert:
+                    env_alert.errors = int(alert["errors"])
+                if "warnings" in alert:
+                    env_alert.warnings = int(alert["warnings"])
+                if "link" in alert:
+                    env_alert.link = alert["link"]
+            env_alerts.append(env_alert)
+    return env_alerts
+
+
+def get_admin_alerts(layout: dict, env_admin_map: Dict[str, dict]) -> List[Alert]:
+    admin_alerts: List[Alert] = []
+    if "admin" not in layout:
+        return admin_alerts
+
+    admin = layout["admin"]
+    for component in admin["componentTypes"]:
+        admin_alert: Alert = Alert("", "", "", component["displayName"], 0, 0, "")
+        if component["id"] in env_admin_map:
+            alert = env_admin_map[component["id"]]
+            if "errors" in alert:
+                admin_alert.errors = int(alert["errors"])
+            if "warnings" in alert:
+                admin_alert.warnings = int(alert["warnings"])
+            if "link" in alert:
+                admin_alert.link = alert["link"]
+        admin_alerts.append(admin_alert)
+    return admin_alerts
+
+
+def parse_arguments(argv: List[str]) -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("user", metavar="USER", help="")
+    parser.add_argument("password", metavar="PASSWORD", help="")
+    parser.add_argument("url", metavar="URL", help="")
+    parser.add_argument("host", metavar="HOST", help="")
+    parser.add_argument(
+        "--debug", action="store_true", help="Debug mode: let Python exceptions raise through"
+    )
+    return parser.parse_args(argv)
+
+
+def get_state(layout_response_code: int, summary_response_code: int, data) -> str:
+    state: str = "up"
+    if layout_response_code == 401 or summary_response_code == 401:
+        state = "unauth"
+    elif layout_response_code != 200 or summary_response_code != 200:
+        state = "down"
+    elif not data:
+        state = "empty"
+    return state
+
+
+def main(args: Optional[List[str]] = None) -> int:
+    if args is None:
+        cmk.utils.password_store.replace_passwords()
+        args = sys.argv[1:]
+    opt: argparse.Namespace = parse_arguments(args)
+
+    layout_connection: LayoutConnection = LayoutConnection(opt.url)
+    layout_request: FuseRequest = FuseRequest(layout_connection.base_url, opt.user, opt.password)
+    layout_response: LayoutResponse = layout_request.get_layout()
```

Comment:
`layout_response = requests.get(f'{opt.url}/layout', auth=(opt.user, opt.password))`

For error handling: when a special agent returns a non-zero exit code, the first line of the error message written to stderr is shown by the "Check_MK" service and it returns a state CRIT. You can use this fact for your error handling here. Eg.

```
try: 
    layout_response = requests.get(f'{opt.url}/layout', auth=(opt.user, opt.password))
except requests.exceptions.RequestException:
    print('Some useful error text', file=sys.stderr)
    raise SystemExit(1)
```